CHANGELOG.md

```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## Unreleased
 
 * Updated the serialization of `WasmMemoryPersistence`.
+* `AgentBuilder::with_background_dynamic_routing` is no longer `async`.
```

Comment:
```suggestion
* [BREAKING] `AgentBuilder::with_background_dynamic_routing` is no longer `async`.
```

This helps us ensure that the next release will be correctly versioned as 0.40.0.